blockly-ws-search/src/WorkspaceSearch.js

```diff
@@ -162,52 +202,43 @@ export class WorkspaceSearch {
     let textInput = document.createElement('input');
     textInput.type = 'text';
     textInput.setAttribute('placeholder', this.textInputPlaceholder_);
-    textInput.addEventListener('keydown',
-        evt => this.onKeyDown_(/** @type {KeyboardEvent} */ evt));
-    textInput.addEventListener('input', () => this.onInput_());
-    textInput.addEventListener('click',
-        () => this.searchAndHighlight(this.preserveSelected));
     return textInput;
   }
 
   /**
-   * Creates the div that holds all of the search bar actions.
-   * @return {!HTMLDivElement} A div holding search bar actions.
+   * Creates the button used to get the next block in the list.
+   * @return {?HTMLButtonElement} The next button.
    * @protected
    */
-  createActionsDiv_() {
-    const actions = document.createElement('div');
-    Blockly.utils.dom.addClass(actions, 'ws-search-actions');
-    // Add all the buttons for the search bar
-    const upBtn = this.createBtn_('up-btn', 'Find previous',
-        () => this.previous());
-    const downBtn = this.createBtn_('down-btn', 'Find next',
-        () => this.next());
-    actions.append(upBtn);
-    actions.append(downBtn);
-    return actions;
+  createNextBtn_() {
+    return this.createBtn('next-btn', 'Find next');
+  }
+
+  /**
+   * Creates the button used to get the previous block in the list.
+   * @return {?HTMLButtonElement} The previous button.
+   * @protected
+   */
+  createPreviousBtn_() {
+    return this.createBtn('previous-btn', 'Find previous');
   }
 
   /**
    * Creates the button used for closing the search bar.
-   * @return {!HTMLElement} A button for closing the search bar.
+   * @return {?HTMLElement} A button for closing the search bar.
    * @protected
    */
   createCloseBtn_() {
-    return this.createBtn_('close-btn', 'Close search bar',
-        () => this.close());
+    return this.createBtn('close-btn', 'Close search bar');
   }
 
   /**
    * Creates a button for the workspace search bar.
    * @param {string} className The class name for the button.
    * @param {string} text The text to display to the screen reader.
-   * @param {!Function} onClickFn The function to call when the user clicks on
-   *    the button.
    * @return {HTMLButtonElement} The created button.
```

Comment:
!HTMLButtonElement 
I think we can say that this method shouldn't return null, even if overriden.

Reply:
ok, done.